### PR TITLE
SM cascade is viable with gas and more visually interesting

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request

Fixes the check for antinob+hypernob cascade for the total amount of gases from the combined_gas to the environmental one.
Allows more damage to the crystal while the cascade is going
Made the antinob and hypernob heat penalty in similar but opposite amounts (15 anti, -13 hyper)
Added transmit modifiers to both gases (-5 anti, 3 hyper)
Increased the amount of crystal shards that spawns from the explosion to a max of 5 and min 2
Added light breaking and random light on emergency mode + maint emergency access when the cascade occurs 

## Why It's Good For The Game

Due to a miscalculation i didn't consider that the combined_gas var refers to the amount of gases that are removed from the environment (that are later used to do all types of calculations). The issue here is that it removes the 15% of the total environment, so this caused a big issue, instead of being able to "just" need around 20k moles of gas (10k hyper and nob) to achieve a cascade delamination, you would have needed 200k Moles of total gas, which is impossible to gather.

The other changes were made to make the cascade viable in terms of SM heat and gas production plus to make the two gases useful in case of normal use.

The light addition is to have a darker station or red lights all around

## Changelog

:cl:
fix: supermatter cascade gas needed amount was broken and you'd needed around 200k Moles of antinob+hypernob (40% each at least) instead of "just" 20k Moles.
balance: changed how hypernob and antinob interact with the SM: heat capacity made similar (15 anti, -13 hyper), transmit modifier added (-5 anti, 3 hyper)
add: lights now break or set to emergency when the cascade starts, maintenance is set to emergency access
/:cl:

### CODE REVIEW

**Comment by Pickle-Coding** [comment]
Why would hypernoblium have a positive transmit modifier? It has a heat penalty of -13, as well as high specific heat, which could easily nullify the difficulty of a lot of gases even at low composition. Sounds a bit strong.

**Comment by Ghilker** [comment]
> Why would hypernoblium have a positive transmit modifier? It has a heat penalty of -13, as well as high specific heat, which could easily nullify the difficulty of a lot of gases even at low composition. Sounds a bit strong.
> 
> Even freon has a higher heat penalty, and completely stops power production at 3% composition.

It's fine, hypernob is not easy to get in high amounts from engies and we might see more of it made for the SM

File: code/modules/power/supermatter/supermatter_delamination.dm [hunk 1]
```diff
@@ -173,13 +173,21 @@
...
 	pick_rift_location()
 	warn_crew()
 	supermatter_turf.ChangeTurf(/turf/closed/indestructible/supermatter_wall)
-	for(var/i in 1 to rand(1,3))
+	for(var/i in 1 to rand(2,5))
 		var/turf/crystal_cascade_location = get_turf(pick(GLOB.generic_event_spawns))
 		crystal_cascade_location.ChangeTurf(/turf/closed/indestructible/supermatter_wall)
 
+/**
+ * Adds a bit of spicyness to the cascade by breaking lights and turning emergency maint access on
```
Comment (tralezab) [inline code/modules/power/supermatter/supermatter_delamination.dm:185]:
```suggestion
 * Adds a bit of spiciness to the cascade by breaking lights and turning emergency maint access on
```

**Review by tralezab — CHANGES_REQUESTED** [review]
HE SPELLED THE WORD WRONG

**Pull request merged**